charts/kubernetes-external-secrets/templates/deployment.yaml

```diff
@@ -0,0 +1,54 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "kubernetes-external-secrets.fullname" . }}
+  labels:
+    app.kubernetes.io/name: {{ include "kubernetes-external-secrets.name" . }}
+    helm.sh/chart: {{ include "kubernetes-external-secrets.chart" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ include "kubernetes-external-secrets.name" . }}
+      app.kubernetes.io/instance: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ include "kubernetes-external-secrets.name" . }}
+        app.kubernetes.io/instance: {{ .Release.Name }}
+      annotations:
+{{- if .Values.podAnnotations }}
+{{ toYaml .Values.podAnnotations | indent 8 }}
+{{- end }}
+    spec:
+      serviceAccountName: {{ template "kubernetes-external-secrets.fullname" . }}
+      containers:
+        - name: {{ .Chart.Name }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          resources:
+            {{- toYaml .Values.resources | nindent 12 }}
+          env:
+            - name: AWS_REGION
+              value: "{{ .Values.deployment.aws.region }}"
+            - name: EVENTS_INTERVAL_MILLISECONDS
+              value: "{{ .Values.deployment.interval.events }}"
+            - name: POLLER_INTERVAL_MILLISECONDS
+              value: "{{ .Values.deployment.interval.poller }}"
+{{- if .Values.extraEnv }}
```

Comment:
Let's just use `.Values.env`.

Reply:
Also, prefer `nindent` over `indent`, which is available since helm 2.7: https://github.com/helm/helm/issues/3854 - it makes templates easier to read and maintain.

Reply:
While its not in the Helm best practice guide, using `extraEnv` and `extraArgs` is a (I believe) a fairly standard approach to expose a consistent place to allow additional values not explicitly required or covered by the chart author (https://github.com/helm/charts/search?q=extraEnv&unscoped_q=extraEnv)

Then by setting the env variables that *must* be provided in the `deployment.yaml` container spec and `values.yaml`, we're guiding the user to what needs to be looked at?

Reply:
Yes I understand your point. However, I do not see, especially at this moment, any use case for it: when do we actually need to set an "extra env" beside standard env stuff such as `AWS_*` and `INTERVAL_*` ?
Finally, if there is any, I think it's still ok to simply add it to `.Values.env` to have 1:1 mapping. It's just more clear. What do you think?

Reply:
Yeah fair point - how about we follow the helm chartmuseum's own approach?

https://github.com/helm/charts/blob/master/stable/chartmuseum/templates/deployment.yaml#L39-L43
https://github.com/helm/charts/blob/master/stable/chartmuseum/values.yaml#L10-L15

That way the `values.yaml` will be a simpler;
```
env:
  AWS_REGION: "us-west-2"
  EVENTS_INTERVAL_MILLISECONDS: 60000
  POLLER_INTERVAL_MILLISECONDS: 10000
```

Reply:
I am not sure I understand what's different here versus what I previously suggested.

Reply:
You can also commit and I'll review again.

Reply:
👍 Theres two approaches to setting env vars in `values.yaml` I guess - the alternative would be something like;
```
env:
  - name: "AWS_REGION"
    value: "us-west-2"
```
I'll push through a new commit with the Chart Museum method